### PR TITLE
Added description of quantization_config

### LINKED ISSUE #31132: Missing description/documentation about quantization_config
Descripion/Documentation about quantization_config parameter is missing in `replace_with_bnb_linear` method at src/transformers/integrations/bitsandbytes.py

### PR DESCRIPTION
**Pull request by vamsivallepu**
Added missing description about quantization_config in replace_with_bnb_linear for better readability.

# What does this PR do?

Fixes #31132 (issue)

## Before submitting
- [x] This PR fixes a typo or improves the docs (you can dismiss the other checks if that's the case).
- [ ] Did you read the [contributor guideline](https://github.com/huggingface/transformers/blob/main/CONTRIBUTING.md#create-a-pull-request),
      Pull Request section?
- [ ] Was this discussed/approved via a Github issue or the [forum](https://discuss.huggingface.co/)? Please add a link
      to it if that's the case.
- [ ] Did you make sure to update the documentation with your changes? Here are the
      [documentation guidelines](https://github.com/huggingface/transformers/tree/main/docs), and
      [here are tips on formatting docstrings](https://github.com/huggingface/transformers/tree/main/docs#writing-source-documentation).
- [ ] Did you write any new necessary tests?

## Who can review?

Anyone in the community is free to review the PR once the tests have passed. Feel free to tag
members/contributors who may be interested in your PR.

### CODE REVIEW

**Comment by LysandreJik** [comment]
@younesbelkada @SunMarc

File: src/transformers/integrations/bitsandbytes.py [hunk 1]
```diff
@@ -243,6 +243,10 @@ def replace_with_bnb_linear(model, modules_to_not_convert=None, current_key_name
             An array to track the current key of the recursion. This is used to check whether the current key (part of
             it) is not in the list of modules to not convert (for instances modules that are offloaded to `cpu` or
             `disk`).
+        quantization_config ('transformers.utils.quantization_config.BitsAndBytesConfig'):
```
Comment (younesbelkada) [inline src/transformers/integrations/bitsandbytes.py:246]:
```suggestion
        quantization_config ('~BitsAndBytesConfig'):
```
This should work in our docs

**Review by younesbelkada — APPROVED** [review]
LGTM thanks !

**Comment by younesbelkada** [comment]
Can you run the styling checks? `make fixup`

**Review by SunMarc — APPROVED** [review]
LGTM !

**Comment by vamsivallepu** [comment]
@SunMarc all tests passed now. Can we merge this now?

**Review by younesbelkada — APPROVED** [review]
Thanks again !

**Pull request merged**